src/modules/custodian/custodian.service.test.ts

```diff
@@ -93,8 +171,160 @@ describe('CustodianService', () => {
+        .mockReturnValueOnce(
+          httpResponseGenerator([
+            {
+              Started: valueGenerator.dateTimeString(),
+              Completed: CUSTODIAN.EMPTY_DATE,
+              Failed: false,
+            },
+          ]),
+        );
+      const response = await service.getApiResponseIfFolderInCustodian(folderParentId, folderName, res);
+
+      expect(res.status).toHaveBeenCalledWith(202);
+      expect(response).toEqual({ folderName, status: ENUMS.FOLDER_STATUSES.CUSTODIAN_JOB_STARTED });
+    });
+
+    it('returns response with status CUSTODIAN_JOB_COMPLETED if custodian job has completion date', async () => {
+      when(cacheManagerGet).calledWith(cacheId).mockReturnValueOnce({ requestId: custodianRequestId, status: ENUMS.FOLDER_STATUSES.CUSTODIAN_JOB_STARTED });
+      when(httpServicePost)
+        .calledWith(...expectedHttpServiceReadJobArgs)
+        .mockReturnValueOnce(
+          httpResponseGenerator([
+            {
+              Started: valueGenerator.dateTimeString(),
+              Completed: valueGenerator.dateTimeString(),
+              Failed: false,
+            },
+          ]),
+        );
+      const response = await service.getApiResponseIfFolderInCustodian(folderParentId, folderName, res);
+
+      expect(res.status).toHaveBeenCalledWith(200);
+      expect(response).toEqual({ folderName, status: ENUMS.FOLDER_STATUSES.CUSTODIAN_JOB_COMPLETED });
+    });
+
+    it('returns response with status CUSTODIAN_JOB_FAILED if custodian job failed flag is set to true', async () => {
+      when(cacheManagerGet).calledWith(cacheId).mockReturnValueOnce({ requestId: custodianRequestId, status: ENUMS.FOLDER_STATUSES.CUSTODIAN_JOB_STARTED });
+      when(httpServicePost)
+        .calledWith(...expectedHttpServiceReadJobArgs)
+        .mockReturnValueOnce(
+          httpResponseGenerator([
+            {
+              Started: valueGenerator.dateTimeString(),
+              Completed: valueGenerator.dateTimeString(),
+              Failed: true,
+            },
+          ]),
+        );
+      const response = await service.getApiResponseIfFolderInCustodian(folderParentId, folderName, res);
+
+      expect(res.status).toHaveBeenCalledWith(500);
+      expect(response).toEqual({ folderName, status: ENUMS.FOLDER_STATUSES.CUSTODIAN_JOB_FAILED });
+      expect(loggerWarn).not.toHaveBeenCalled();
+    });
+
+    it('returns null if custodian job failed in previous run, and we will allow repeated call', async () => {
+      when(cacheManagerGet).calledWith(cacheId).mockReturnValueOnce({ requestId: custodianRequestId, status: ENUMS.FOLDER_STATUSES.CUSTODIAN_JOB_FAILED });
+
+      const response = await service.getApiResponseIfFolderInCustodian(folderParentId, folderName, res);
+
+      expect(res.status).not.toHaveBeenCalled();
+      expect(response).toBeNull();
+      expect(loggerWarn).toHaveBeenCalledWith('Found cache with failed Custodian job for folder creation. Lets try to create new job.');
```

Comment:
`Lets try to create new job.` -> `Let's try to create new job`

Reply:
Thanks, changed